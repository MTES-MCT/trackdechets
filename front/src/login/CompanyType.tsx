import { FieldArray, FieldProps } from "formik";
import React, { InputHTMLAttributes } from "react";
import Tooltip from "common/components/Tooltip";
import { CompanyType } from "../generated/graphql/types";

export const COMPANY_TYPES = [
  {
    value: CompanyType.Producer,
    label: "Producteur de déchets",
  },
  {
    value: CompanyType.Collector,
    label: "Installation de Transit, regroupement ou tri de déchets",
    helpText:
      "Installations sur lesquelles sont regroupés, triés ou en transit les déchets dangereux et/ou non dangereux - installations relevant des rubriques suivantes de la nomenclature ICPE:  2711, 2713, 2714, 2715, 2716, 2718, 2719, 2731, 2792-1, 2793-1, 2793-2, 2797-1, 2798.",
  },
  {
    value: CompanyType.WasteCenter,
    label:
      "Installation de collecte de déchets apportés par le producteur initial",
    helpText:
      "Déchetteries et installations relevant de la rubrique 2710 de la nomenclature ICPE",
  },
  {
    value: CompanyType.WasteVehicles,
    label:
      "Installation d'entreposage, dépollution, démontage, découpage de VHU",
    helpText:
      "Casse automobile, installations d'entreposage, dépollution, démontage de tout type de véhicules hors d'usage - installations relevant de la rubrique 2712 de la nomenclature ICPE",
  },
  {
    value: CompanyType.Wasteprocessor,
    label: "Installation de traitement",
    helpText:
      "Installations sur lesquelles sont traités les déchets, et relevant des rubriques suivantes de la nomenclature ICPE :  2720, 2730, 2740, 2750, 2751, 2752, 2760, 2770, 2771, 2780, 2781, 2782, 2790, 2791, 2792-2, 2793-3, 2794, 2795, 2797-2 et 3510, 3520, 3531, 3532, 3540, 3550, 3560.",
  },
  {
    value: CompanyType.Transporter,
    label: "Transporteur",
    helpText:
      "Transporteur de déchets (dangereux) disposant d'un récépissé préfectoral",
  },
  {
    value: CompanyType.Trader,
    label: "Négociant",
    helpText:
      "Négociant, prenant part à la relation producteur / traiteur, disposant d'un récépissé préfectoral",
  },
  {
    value: CompanyType.EcoOrganisme,
    label: "Éco-organisme",
    helpText:
      "Société prenant en charge la gestion des déchets, dans le cadre de la REP (Responsabilité élargie du producteur)",
  },
];

export default function CompanyTypeField({
  field: { name, value },
  id,
  label,
  ...props
}: FieldProps & { label: string } & InputHTMLAttributes<HTMLInputElement>) {
  return (
    <FieldArray
      name={name}
      render={arrayHelpers => (
        <fieldset className={name + "-fieldset"}>
          {COMPANY_TYPES.map((companyType, idx) => (
            <div key={idx}>
              <label
                style={{ display: "inline-block" }}
                key={companyType.value}
              >
                <input
                  type="checkbox"
                  name={name}
                  className="td-checkbox"
                  value={companyType.value}
                  checked={value.includes(companyType.value)}
                  onChange={e => {
                    if (e.target.checked) {
                      arrayHelpers.push(companyType.value);
                    } else {
                      const idx = value.indexOf(companyType.value);
                      arrayHelpers.remove(idx);
                    }
                  }}
                />
                {companyType.label}
                <Tooltip msg={companyType.helpText} />
              </label>
            </div>
          ))}
        </fieldset>
      )}
    />
  );
}
