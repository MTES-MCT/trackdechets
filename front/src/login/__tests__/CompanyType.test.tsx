import CompanyType from "../CompanyType";
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";

describe("<CompanyType />", () => {
  const props = { field: { name: "my-field", value: [true] } };
  it("CompanyType renders correctly", () => {
    const { container, debug, queryByText } = render(
      <CompanyType {...props} />
    );

    expect(queryByText("Producteur de d√©chets")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
