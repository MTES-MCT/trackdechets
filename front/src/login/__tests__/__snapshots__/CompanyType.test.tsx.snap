// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyType /> CompanyType renders correctly 1`] = `
<fieldset
  class="companyTypes-fieldset"
>
  <div>
    <label
      style="display: inline-block;"
    >
      <input
        class="td-checkbox"
        name="companyTypes"
        type="checkbox"
        value="PRODUCER"
      />
      Producteur de déchets
    </label>
  </div>
  <div>
    <label
      style="display: inline-block;"
    >
      <input
        class="td-checkbox"
        name="companyTypes"
        type="checkbox"
        value="COLLECTOR"
      />
      Installation de Transit, regroupement ou tri de déchets
      <span
        class="tooltip__container"
      >
        <span
          class="tooltip__trigger"
          role="button"
        >
          <svg
            class="tooltip__icon"
            fill="currentColor"
            height="1em"
            role="icon"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
            />
          </svg>
        </span>
      </span>
    </label>
  </div>
  <div>
    <label
      style="display: inline-block;"
    >
      <input
        class="td-checkbox"
        name="companyTypes"
        type="checkbox"
        value="WASTE_CENTER"
      />
      Installation de collecte de déchets apportés par le producteur initial
      <span
        class="tooltip__container"
      >
        <span
          class="tooltip__trigger"
          role="button"
        >
          <svg
            class="tooltip__icon"
            fill="currentColor"
            height="1em"
            role="icon"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
            />
          </svg>
        </span>
      </span>
    </label>
  </div>
  <div>
    <label
      style="display: inline-block;"
    >
      <input
        class="td-checkbox"
        name="companyTypes"
        type="checkbox"
        value="WASTE_VEHICLES"
      />
      Installation d'entreposage, dépollution, démontage, découpage de VHU
      <span
        class="tooltip__container"
      >
        <span
          class="tooltip__trigger"
          role="button"
        >
          <svg
            class="tooltip__icon"
            fill="currentColor"
            height="1em"
            role="icon"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
            />
          </svg>
        </span>
      </span>
    </label>
  </div>
  <div>
    <label
      style="display: inline-block;"
    >
      <input
        class="td-checkbox"
        name="companyTypes"
        type="checkbox"
        value="WASTEPROCESSOR"
      />
      Installation de traitement
      <span
        class="tooltip__container"
      >
        <span
          class="tooltip__trigger"
          role="button"
        >
          <svg
            class="tooltip__icon"
            fill="currentColor"
            height="1em"
            role="icon"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
            />
          </svg>
        </span>
      </span>
    </label>
  </div>
  <div>
    <label
      style="display: inline-block;"
    >
      <input
        class="td-checkbox"
        name="companyTypes"
        type="checkbox"
        value="TRANSPORTER"
      />
      Transporteur
      <span
        class="tooltip__container"
      >
        <span
          class="tooltip__trigger"
          role="button"
        >
          <svg
            class="tooltip__icon"
            fill="currentColor"
            height="1em"
            role="icon"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
            />
          </svg>
        </span>
      </span>
    </label>
  </div>
  <div>
    <label
      style="display: inline-block;"
    >
      <input
        class="td-checkbox"
        name="companyTypes"
        type="checkbox"
        value="TRADER"
      />
      Négociant
      <span
        class="tooltip__container"
      >
        <span
          class="tooltip__trigger"
          role="button"
        >
          <svg
            class="tooltip__icon"
            fill="currentColor"
            height="1em"
            role="icon"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
            />
          </svg>
        </span>
      </span>
    </label>
  </div>
</fieldset>
`;
